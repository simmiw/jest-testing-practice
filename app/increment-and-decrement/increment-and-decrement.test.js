import IncrementAndDecrement from "./page";
import { render, fireEvent, screen } from "@testing-library/react";

describe("IncrementAndDecrement", () => {
  it("should render buttons", async () => {
    const sut = render(<IncrementAndDecrement/>);
    expect(sut.container).toMatchSnapshot();
  });

  it("should render the Increment button, when clicked", () => {
    render(<IncrementAndDecrement />);
    fireEvent.click(screen.getByText("Increment"));
  });

  it("should render the Decrement button, when clicked", () => {
    render(<IncrementAndDecrement />);
    fireEvent.click(screen.getByText("Decrement"));
  });
});
