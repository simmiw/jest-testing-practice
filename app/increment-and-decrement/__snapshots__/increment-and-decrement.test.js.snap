// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncrementAndDecrement should render buttons 1`] = `
<div>
  <h1
    class="text-center font-bold my-5 "
  >
     Increment and Decrement
  </h1>
  <div
    class="text-center m-10 "
  >
    <button
      class=" font-bold py-2 px-4 text-white font-bold  rounded-full bg-blue-400"
    >
      Increment
    </button>
    <div>
      0
    </div>
    <h2>
      You are at intital state of counter!
    </h2>
    <button
      class=" font-bold py-2 px-4 text-white font-bold  rounded-full bg-blue-400"
    >
      Decrement
    </button>
  </div>
</div>
`;
